### PR TITLE
espressif_3.0

### PR DESCRIPTION
**Pull request by kotovasia5120**
Fix for compiling espressif_3.0

### CODE REVIEW

File: h4/h4.ino [hunk 1]
```diff
@@ -1862,9 +1862,8 @@ void updateAsyncTimerSettings() {
   setDir(getAsyncAxis(), dupr > 0);
 
   // dupr can change while we're in async mode, keep updating timer frequency.
-  timerAlarmWrite(async_timer, getTimerLimit(), true);
   // without this timer stops working if already above new limit
-  timerWrite(async_timer, 0);
```
Comment (kachurovskiy) [inline h4/h4.ino:1867]: Why do you remove this line? timerAlarm() doesn't seem to reset the timer as per docs.

Comment (kachurovskiy) [reply h4/h4.ino:1867]: Order of statements here is important, we first configure the timer and then reset it's value to 0.

```
  // dupr can change while we're in async mode, keep updating timer frequency.
  timerAlarm(async_timer, getTimerLimit(), true, 0);
  // without this timer stops working if already above new limit
  timerWrite(async_timer, 0);
```

Comment (kotovasia5120) [reply h4/h4.ino:1867]: It was my mistake.

File: h4/h4.ino [hunk 2]
```diff
@@ -1666,7 +1666,7 @@ void setup() {
   isOn = false;
   savedDupr = dupr = pref.getLong(PREF_DUPR);
   motionMutex = xSemaphoreCreateMutex();
-  savedStarts = starts = min(STARTS_MAX, max(1, pref.getInt(PREF_STARTS)));
+  savedStarts = starts = min(static_cast<long>(STARTS_MAX), max(static_cast<int32_t>(1), pref.getInt(PREF_STARTS)));
```
Comment (kotovasia5120) [inline h4/h4.ino:1669]: This design is difficult for me. I could be wrong.

Comment (kachurovskiy) [reply h4/h4.ino:1669]: Can you just use `(int32_t) 1` style of conversion?

Comment (kotovasia5120) [reply h4/h4.ino]: So?
https://github.com/kachurovskiy/nanoels/pull/210/commits/8d1285259bee776ec448dcc5d7a11206e350ce1a

**Comment by kachurovskiy** [comment]
Can you please test that async mode is working as intended with your latest changes? Also try lowering/increasing pitch while ASYNC mode is ON

**Review by tp1845 — COMMENTED** [review]
no issue

**Pull request merged**